cEP-0010.md

```diff
@@ -136,10 +138,37 @@ Here is the list of aspects to be implemented along with their corresponding
 
 - `Root.Formatting`: This aspect checks on the format (the structure) of your
   codebase.
-- `Root.Formatting.Quotation`: This aspect checks on the quotations being used
-  in your files given `Root.Formatting.Quotation.preferred_quotation`.
+- `Root.Formatting.Quotation`: This aspect represents the quotation marks
+  being used in the files written in languages with multiple string delimiters,
+  which is common in recent languages, and excludes C-ish languages.
 - `Root.Formatting.Quotation.preferred_quotation`: This taste represents the
-  user's preferred quotation.
+  preferred quotation marks for the selected files. It ensures that every string
+  contains the selected style of quotes.
+  For example, `'This is single quoted'` is changed to `"This is single quoted"`
+  if `preferred_quotation` is `"`.
+- `Root.Formatting.Quotation.escape_delimiter`: This taste ensures that
+  selected quotes are strictly followed even in cases where other quotation
+  marks are present inside string containing the selected style of quotes.
+  This is done by replacing other quotes with selected quotes and escaping
+  them wherever necessary.
+  For example, `"Hey 'coala'"` is changed to `"Hey \"coala\""` by setting
+  `preferred_quotation` as `"` and `escape_delimiter` as `True`.
+  Similarly, `'Hey "coala"'` is changed to `"Hey \"coala\""` with the
+  same settings as mentioned above.
+  For languages which do not support escaping quotes like `\"`, a warning
+  is shown. For such languages, taste `alternate_delimiter` is used
```

Comment:
this taste shouldnt talk about other tastes.  there will be *lots* of other tastes which handle different languages, so dont mention any of them.

Please add language definition and aspects support for escape sequences, as described at https://en.wikipedia.org/wiki/Delimiter#Escape_sequence and https://en.wikipedia.org/wiki/Escape_sequence  This is the most commonly available mechanism for delimiter clashes - most languages support this, but newer languages provide less ugly syntax.  But we need it, as it provides a baseline functionality that ensures we can support most languages, even if it isnt pretty.

Also please add language definition and aspects support for here documents, as described at https://en.wikipedia.org/wiki/Here_document .  These are also quite commonly available, and this is relevant as they are another form of avoiding delimiter collision as the string delimiter is configurable. https://en.wikipedia.org/wiki/Delimiter#Configurable_alternate_quoting_delimiters  While the cEP should define this taste, this one will be the hardest to implement, so implementation might be a stretch goal for your GSoC project.  A very important point is that while many languages support here documents, it is usually for multiline strings, with very few languages allowing single line here documents.

https://en.wikipedia.org/wiki/String_literal#Constructor_functions and string literal concatenation are more complicated variations, and I suggest that we put a note in the cEP that these should be added at a later time.  It is too hard to design these (now), and they are less valuable than escape sequences and `here` strings/documents.

Reply:
I think you are asking a lot from this GSoC project and it's somewhat off a bit. Think about it, this project aims more at creating a sound working functionality of bear by making use of aspects. More stress is on converting bears to aspects rather than providing extra functionalities in one go.

I think we should restrict this project to what it is. Common string literals in languages is what we should stick to. Here documents and the rest could be implemented in some other project, if the proposed functionality of this aspect proves to be a success. Moreover, we can stretch `QuotesBear` beyond limits as quotations are the most common thing in any language. 

Here, instead of extending the functionality of a particular bear, we should focus on migrating the bear for a limited functionality. This will be more of migrating than creating and implementing aspects and migrating is what the project is all about.

Think about this a bit and do tell me if I am wrong on my part in any of these statements. It's always better to clarify doubts early. :smile:

Reply:
Oh, I am not proposing that we enhance `QuotesBear` to handle these other types of string delimiter systems.  I am asking that we build the aspect and tastes correctly.

These tastes may not be implemented in this project, and I expect that they are not.  However the process of building a taste is only a documentation task.  It only describes the task, and leaves implementation of it to bears.  In the case of Here documents, we dont have any bear which would implement that.  However describing the Here taste is important for conceptually building the aspect correctly, IMO.